### PR TITLE
Correct look task behavior when renaming

### LINKED ISSUE #139: Database not updated when renaming a ComplexTask ?
exopy version: 0.2.0.dev.post11

In this measurement:
[M.meas.ini.zip](https://github.com/Exopy/exopy/files/1942095/M.meas.ini.zip)

if you rename the Test2 task to Test2bis and then try to move it out of Test1, it looks like the name of the measure is not updated correctly in the database.

you get:
```python
Traceback (most recent call last):
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\utils\widgets\
qt_tree_widget.py", line 1072, in dropEvent
    from_index)
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\utils\widgets\
qt_tree_widget.py", line 607, in _delete
    node.delete_child(obj, index)
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\measurement\wo
rkspace\measurement_edition.enaml", line 409, in delete_child
    obj.remove_child_task(index)
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\tasks\tasks\ba
se_tasks.py", line 783, in remove_child_task
    child.unregister_from_database()
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\tasks\tasks\ba
se_tasks.py", line 855, in unregister_from_database
    super(ComplexTask, self).unregister_from_database()
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\tasks\tasks\ba
se_tasks.py", line 263, in unregister_from_database
    self.database.delete_value(self.path, self._task_entry(entry))
  File "D:\Users\hqc\Anaconda2\envs\exopy\lib\site-packages\exopy\tasks\tasks\da
tabase.py", line 252, in delete_value
    raise KeyError(err_str)
KeyError: 'No entry Test2bis_point_number in node root/Test1'

```

### PR DESCRIPTION
**Pull request by lcontami**
The database was not updated.
Fixes #139

### CODE REVIEW

**Comment by lcontami** [comment]
pfff I'll correct the names tomorrow

File: tests/tasks/tasks/logic/test_loop_task.py [hunk 1]
```diff
@@ -204,6 +204,17 @@ def test_subtask_handling(self):
...
+    def test_renaming(self):
+        """Test renaming the task.
+
+        """
+        subtask1 = CheckTask(name='check', database_entries={'val': 1})
+        self.task.task = subtask1
+        assert subtask1.get_from_database('Test_val') == 1
+
+        self.task.name = 'Test2'
+        assert subtask1.get_from_database('Test2_val') == 1
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/logic/test_loop_task.py]: This does not test that we actually called the super method. We need to also check that we renamed the entries of the LoopTask itself and not only of the embedded subtask.

File: exopy/tasks/tasks/logic/loop_task.py [hunk 2]
```diff
@@ -234,3 +234,4 @@ def _post_setattr_name(self, old, new):
         """
         if self.task:
             self.task.name = new
+        super(LoopTask, self)._post_setattr_name(old, new)
```
Comment (MatthieuDartiailh) [inline exopy/tasks/tasks/logic/loop_task.py]: We can use the argument less super here since we do not support Python 2 anymore.

**Comment by lcontami** [comment]
Ready to merge if you are happy madar :)

**Comment by MatthieuDartiailh** [comment]
Did you address my comments @lcontami ? Github does not show any update, maybe you forgot to push ?

File: tests/tasks/tasks/logic/test_loop_task.py (hunk 1 above)
Comment (lcontami) [reply tests/tasks/tasks/logic/test_loop_task.py]: The renaming of the LoopTask entries is already tested in the renaming of a complextask, do you want to retest it ?

Comment (lcontami) [reply tests/tasks/tasks/logic/test_loop_task.py]: (or maybe somewhere else but that's why I did not test it also)

Comment (MatthieuDartiailh) [reply tests/tasks/tasks/logic/test_loop_task.py]: I want to be sure we do call the renaming of the ComplexTask, this was the issue after all. But simply checking one value is fine (with a comment explaining why)

File: tests/tasks/tasks/logic/test_loop_task.py [hunk 3]
```diff
@@ -204,6 +204,18 @@ def test_subtask_handling(self):
...
+        assert self.task.get_from_database('Test_point_number') == 11
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/logic/test_loop_task.py]: If this pass it means we do have a problem !!!

**Comment by MatthieuDartiailh** [comment]
Thanks @lcontami

**Pull request merged**